lxd/project/limits/permissions.go

```diff
@@ -1348,12 +1359,18 @@ func getInstanceLimits(instance api.Instance, keys []string, skipUnset bool) (ma
 			if instance.Type == instancetype.VM.String() {
 				sizeStateValue, ok := device["size.state"]
 				if !ok {
-					// TODO: In case the VMs storage drivers config drive size isn't the default,
-					// the limits accounting will be incorrect.
-					// This applies for the PowerFlex storage driver whose config drive size
-					// is 8 GB as set in the DefaultVMPowerFlexBlockFilesystemSize variable.
-					// See https://github.com/canonical/lxd/issues/12567
-					sizeStateValue = deviceconfig.DefaultVMBlockFilesystemSize
+					driverName := ""
+					poolName := device["pool"]
+					for pool, driver := range storagePoolDrivers {
```

Comment:
Do you need a loop or can you just do this:

```golang
driverName, ok := storagePoolDrivers[poolName]
```

Reply:
Yeh that would be nicer.

Also what happens if the poolName isn't matched to a driver @MggMuggins ?

Reply:
Oops. Adjusted this for better error messages if the root device is missing a `pool` attribute (sounds like this [shouldn't happen](https://github.com/canonical/lxd/blob/main/lxd/instance/instancetype/instance.go#L32)) or the pool doesn't have a driver.

Reply:
@MggMuggins yeah I think the check doesn't hurt as the error message is more usable than `No driver found for pool ""` in case the pool attribute isn't set in the device map.